packages/frontend/src/hooks/useChartColorConfig.ts

```diff
@@ -0,0 +1,108 @@
+import { FeatureFlags, Series } from '@lightdash/common';
+import { useFeatureFlagEnabled } from 'posthog-js/react';
+import { useCallback, useState } from 'react';
+import { EChartSeries } from './echarts/useEchartsCartesianConfig';
+
+export type SeriesLike = EChartSeries | Series;
+
+export const useChartColorConfig = ({
+    colorPalette,
+}: {
+    colorPalette: string[];
+}) => {
+    const [colorMappings] = useState(new Map<string, string>());
+    const useSharedColors = useFeatureFlagEnabled(
+        FeatureFlags.UseSharedColorAssignment,
+    );
+
+    /**
+     * Given the org's color palette, and an identifier, return the color palette value
+     * for said identifier.
+     *
+     * This works by hashing the identifier into an integer value, and then projecting
+     * that value into the org's color space - effectivelly getting a number from 0 to
+     * <number of colors in palette>.
+     *
+     * This is a straight-forward way to handle hashing dimensions into colors, with
+     * two major caveats:
+     *
+     * - We're no longer cycling over colors in the palette, potentially not following
+     *   an intentionally-designed best-neighbor-color approach.
```

Comment:
Not a blocker, just something that could help with this: https://gka.github.io/chroma.js/#scale-colors